src/commands/qr/index/processor.ts

```diff
@@ -391,6 +392,50 @@ export async function viewQR(i: SelectMenuInteraction) {
   return messageOptions
 }
 
+function buildQueryUrl(d: {
+  category: string
+  amount?: string
+  symbol?: string
+  content?: string
+  type?: string
+  id?: string
+  address?: string
+}) {
+  let qrValue = ""
+  const {
+    amount = "",
+    symbol = "",
+    content = "",
+    type = "",
+    category,
+    id = "",
+    address = "",
+  } = d
+
+  const url = new URL(`${API_BASE_URL}/pk-pass`)
+  url.searchParams.append("category", category)
+
+  if (category === "social") {
+    if (type.toLowerCase() === "twitter") {
+      qrValue = `${TWITTER_USER_URL}/${content}`
+    } else if (type.toLowerCase() === "telegram") {
+      qrValue = `${TELEGRAM_USER_URL}/${content}`
+    }
+  } else if (category === "pay") {
+    qrValue = id.split("-")[2]
+  } else if (category.includes("wallet")) {
+    qrValue = address
+  }
+
+  url.searchParams.append("qr_value", qrValue)
+  url.searchParams.append("amount", amount)
+  url.searchParams.append("symbol", symbol)
+  url.searchParams.append("content", content)
+  url.searchParams.append("type", type)
+
+  return url.toString()
+}
```

Comment:
can u revert back to the previous one? I remember a bug when using URL searchparams which may happen to `content`

```
return `${API_BASE_URL}/pk-pass?category=${encodeURIComponent(	  url.searchParams.append("qr_value", qrValue)
    category	  url.searchParams.append("amount", amount)
  )}&qr_value=${encodeURIComponent(qrValue)}&amount=${encodeURIComponent(	  url.searchParams.append("symbol", symbol)
    amount	  url.searchParams.append("content", content)
  )}&symbol=${encodeURIComponent(symbol)}&content=${encodeURIComponent(	  url.searchParams.append("type", type)
    content	
  )}&type=${encodeURIComponent(type)}`
```